xarray/core/nputils.py

```diff
@@ -79,3 +79,56 @@ def array_ne(self, other):
     with warnings.catch_warnings():
         warnings.filterwarnings('ignore', r'elementwise comparison failed')
         return _ensure_bool_is_ndarray(self != other, self, other)
+
+
+def _is_contiguous(positions):
+    """Given a non-empty list, does it consist of contiguous integers?"""
+    previous = positions[0]
+    for current in positions[1:]:
+        if current != previous + 1:
+            return False
+        previous = current
+    return True
```

Comment:
I think we can vectorize this:

```Python
def _is_contiguous(positions):
     """Given a non-empty list, does it consist of contiguous integers?"""
     if (len(positions) == 1) or (np.diff(positions) == 1).all():
         return True
     return False
```

Reply:
Actually, `all()` is defined in the right way for length 0 lists. So this can just be:
```python
def _is_contiguous(positions):
    return (np.diff(positions) == 1).all()
```

This is arguably an improvement for readability (at least for the numpy familiar devs who work on xarray) but there's no *advantage* to using this because `positions` only has one element per indexer, so it almost always a list of only a few items.